WeatherForecastApp/Controller/LocationDetailViewController.swift

```diff
@@ -22,22 +24,27 @@ class LocationDetailViewController: UIViewController {
     @IBOutlet weak var summaryLabel: UILabel!
     @IBOutlet weak var imageView: UIImageView!
     @IBOutlet weak var pageControl: UIPageControl!
-    @IBOutlet weak var collectionView: UICollectionView!
     @IBOutlet weak var tableView: UITableView!
-    //var weatherLocation: WeatherLocation = WeatherLocation(name: "Current Location", latitude: 0.0, longitude: 0.0)
-    //var weatherLocations: [WeatherLocation] = []
-    var weatherDetail: WeatherDetail!
+
+    var weatherLocation: WeatherLocation!
     var locationIndex = 0
+    var locationManager: CLLocationManager!
+    var weatherDetail: WeatherDetail?
     
     override func viewDidLoad() {
         super.viewDidLoad()
-       clearUserInterface()
-        
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        clearUserInterface()
         tableView.delegate = self
         tableView.dataSource = self
-//        collectionView.delegate = self
-//        collectionView.dataSource = self
-//        
+        
+        if locationIndex == 0 {
```

Comment:
this is generally a hack and this is not a scalable solution. You would ideally want to get new Location data when a user opens the app and goes to this screen. In some cases , in production apps we have a cache that stores data and updates at a specific interval.